### ISSUE TITLE
Avoid duplicate diagnostics from D400 and D415 when both rules are enabled

### ISSUE THREAD

**Issue by vivodi** [astral-sh/ruff#27495]
## Summary

`D415` (`missing-terminal-punctuation`) is effectively a superset of `D400` (`missing-trailing-period`) for docstrings that do not end with any terminal punctuation.

When both rules are enabled, Ruff reports two diagnostics for the same underlying problem:

```python
def foo():
    """Return the value"""
```

Current output:

```
D400 First line should end with a period
D415 First line should end with a period, question mark, or exclamation point
```

This results in duplicate diagnostics for a single issue.

## Expected behavior

When both `D400` and `D415` are enabled, Ruff should emit only `D400` for docstrings that lack terminal punctuation entirely, since satisfying `D400` also satisfies `D415`.

## Alternative approach

**Another possible solution** would be to merge these two rules into a single rule with a configuration option controlling the allowed terminal punctuation.

For example:

* `period` (current `D400` behavior)
* `terminal-punctuation` (current `D415` behavior)

This would eliminate the overlap while preserving both styles as configurable options.

**Comment by ntBre** [comment]
This was previously brought up in https://github.com/astral-sh/ruff/issues/8469, which was closed in favor of #1774. I'll add this as a sub-issue of #1774. Codex said that setting your [convention](https://docs.astral.sh/ruff/settings/#lint_pydocstyle_convention) is supposed to fix this, though. D415 is enabled for the numpy and PEP 257 conventions, while D400 is enabled for the Google convention. It still probably makes sense to fold the two rules into one as part of recategorization.

**Comment by Ruu-0** [comment]
Hi! I reproduced this locally and have a fix with regression tests ready.

Since this issue isn't labeled `good first issue` or `help wanted`, would you be open to a PR for this?
